astropy/time/core.py

```diff
@@ -2457,6 +2496,40 @@ def _make_value_equivalent(self, item, value):
                                  'object: {}'.format(err))
         return value
 
+    def isclose(self, other, atol=None, rtol=0.0):
+        """Returns a boolean or boolean array where two TimeDelta objects are
+        element-wise equal within a time tolerance.
+
+        This effectively evaluates the expression below::
+
+          abs(self - other) <= atol + rtol * abs(other)
+
+        Parameters
+        ----------
+        other : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`
+            Quantity or TimeDelta object for comparison.
+        atol : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`
+            Absolute tolerance for equality with units of time (e.g. ``u.s`` or
+            ``u.day``). Default is one bit in the 128-bit JD time representation,
+            equivalent to about 20 picosecs.
+        rtol : float
+            Relative tolerance for equality
+        """
+        try:
+            other_day = other.to_value(u.day)
+        except Exception as err:
```

Comment:
I'd follow `Time.isclose()` and do `atol_day = atol.to_value(u.day)` inside here (and adjust the error message), with the `atol is None` statement moved in front.